iam.tf

```diff
@@ -318,17 +488,586 @@ data "aws_iam_policy_document" "doublecloud_permissions" {
+    sid    = "AutoscalingAllowAllControlplaneEKS"
+    effect = "Allow"
+    actions = [
+      "autoscaling:*",
+    ]
+    resources = ["*"]
+    condition {
+      test     = "StringEquals"
+      values   = ["doublecloud-platform"]
+      variable = "aws:ResourceTag/eks:nodegroup-name"
+    }
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "autoscaling:Describe*",
+      "autoscaling:Get*",
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    sid    = "EC2AllowAllWithinDoubleCloudVPC"
+    effect = "Allow"
+    actions = [
+      "ec2:*"
+    ]
+    resources = ["*"]
+    condition {
+      test     = "StringEquals"
+      values   = [aws_vpc.doublecloud.arn]
+      variable = "ec2:Vpc"
+    }
+  }
+
+  statement {
+    sid    = "EC2AllowAllDoubleCloudVPC"
+    effect = "Allow"
+    actions = [
+      "ec2:*"
+    ]
+    resources = [aws_vpc.doublecloud.arn]
+  }
+
+  statement {
+    sid    = "EC2AllowAllDoubleCloud"
+    effect = "Allow"
+    actions = [
+      "ec2:*"
+    ]
+    resources = [
+      "arn:aws:ec2:${local.region}:${local.account_id}:*/*",
+      "arn:aws:ec2:${local.region}::*/*",
+    ]
+    condition {
+      test     = "StringEquals"
+      values   = ["true"]
+      variable = "aws:ResourceTag/atDoubleCloud"
+    }
+  }
+
+  statement {
+    sid    = "EC2AllowAllDoubleCloudNamed"
+    effect = "Allow"
+    actions = [
+      "ec2:*"
+    ]
+    resources = ["*"]
+    condition {
+      test     = "StringLike"
+      values   = ["DoubleCloud-Airflow*"]
+      variable = "aws:ResourceTag/Name"
+    }
+  }
+
+  statement {
+    sid = "EKSFullAccessDoubleCloud"
+    actions = [
+      "eks:*",
+    ]
+    resources = ["arn:aws:eks:${local.region}:${local.account_id}:*/DoubleCloud-Airflow-*"]
+  }
+
+  statement {
+    sid    = "ElasticLoadBalancingAllowAllDoubleCloud"
+    effect = "Allow"
+    actions = [
+      "elasticloadbalancing:*"
+    ]
+    resources = ["*"]
+    condition {
+      test     = "StringEquals"
+      values   = ["true"]
+      variable = "aws:ResourceTag/AtDoubleCloud"
+    }
+  }
+
+  statement {
+    sid    = "DescribeElasticLoadBalancing"
+    effect = "Allow"
+    actions = [
+      "elasticloadbalancing:Describe*",
+      "elasticloadbalancing:Get*"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    sid = "EKSAllowPassRolesDoubleCloud"
+    actions = [
+      "iam:PassRole",
+    ]
+    effect    = "Allow"
+    resources = ["arn:aws:iam::${local.account_id}:role/*"]
+    condition {
+      test     = "StringEquals"
+      variable = "iam:PassedToService"
+      values   = ["eks.amazonaws.com"]
+    }
+  }
+
+  statement {
+    sid    = "EKSNodeGroupIAMPolicyDoubleCloud"
+    effect = "Allow"
+    actions = [
+      "iam:GetRole",
+      "iam:ListAttachedRolePolicies",
+    ]
+    resources = ["arn:aws:iam::${local.account_id}:role/DoubleCloud/*"]
+  }
+
+  statement {
+    sid       = "EKSAllowCreateServiceLinkedRole"
+    effect    = "Allow"
+    resources = ["*"]
```

Comment:
We allow the creation of any service-linked role that can lead to misuse. Can we add some filters here? Maybe add tag-based restrictions?

Reply:
Negative, the condition is below? i.e we can only create ServiceLinkedRoles for the services defined here -> https://github.com/doublecloud/terraform-aws-doublecloud-byoc/pull/6/files#diff-2bb47ab35b4fcfacd8f871bd1f4bd594f5d1c2a89c0eb861c0e305175a0ef4ecR669-R680

This is required as per AWS upstream policy